pkg/api/routes.go

```diff
@@ -179,17 +179,11 @@ func (rh *RouteHandler) SetupRoutes() {
 	// gql playground
 	gqlPlayground.SetupGQLPlaygroundRoutes(prefixedRouter, rh.c.StoreController, rh.c.Log)
 
-	// setup extension routes
-	if rh.c.Config != nil {
-		// This logic needs to be reviewed, it should depend on build options
-		// not the general presence of the extensions in config
-		if rh.c.Config.Extensions == nil {
-			// minimal build
-			prefixedRouter.HandleFunc("/metrics", rh.GetMetrics).Methods("GET")
-		} else {
-			// extended build
-			ext.SetupMetricsRoutes(rh.c.Config, rh.c.Router, authHandler, rh.c.Log)
-		}
+	if rh.c.Config.BinaryType == "minimal" {
```

Comment:
We don't need to make this check here if the route is enabled by ext.SetupMetricsRoutes(), which will be selected from the correct file at build time.